### PR TITLE
test(v1.x): expand apphash test with all state machine msgs

### PR DESCRIPTION
**Pull request by ninabarbakadze**
## Overview

Consistent Apphash test on v1

### CODE REVIEW

File: app/consistent_apphash_test.go [hunk 1]
```diff
@@ -3,134 +3,303 @@ package app_test
...
 
 	"github.com/celestiaorg/celestia-app/app"
 	"github.com/celestiaorg/celestia-app/app/encoding"
 	"github.com/celestiaorg/celestia-app/pkg/appconsts"
-	appns "github.com/celestiaorg/celestia-app/pkg/namespace"
 	"github.com/celestiaorg/celestia-app/pkg/user"
 	testutil "github.com/celestiaorg/celestia-app/test/util"
 	"github.com/celestiaorg/celestia-app/test/util/blobfactory"
+	"github.com/celestiaorg/celestia-app/test/util/testfactory"
+
```
Comment (rootulp) [inline app/consistent_apphash_test.go:15]: does your editor automatically add these newlines between imports? They seem unecessary to me. If all the imports are in one block, then golangci-lint will order them correctly.

```suggestion
```

File: app/consistent_apphash_test.go [hunk 2]
```diff
@@ -3,134 +3,303 @@ package app_test
...
+	appns "github.com/celestiaorg/celestia-app/pkg/namespace"
 	"github.com/cosmos/cosmos-sdk/codec"
 	"github.com/cosmos/cosmos-sdk/crypto/hd"
 	"github.com/cosmos/cosmos-sdk/crypto/keyring"
 	"github.com/cosmos/cosmos-sdk/crypto/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+
```
Comment (rootulp) [inline app/consistent_apphash_test.go:22]:
```suggestion
```

File: app/consistent_apphash_test.go [hunk 3]
```diff
@@ -3,134 +3,303 @@ package app_test
...
+	blobtypes "github.com/celestiaorg/celestia-app/x/blob/types"
+	"github.com/cosmos/cosmos-sdk/x/authz"
 	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	crisisTypes "github.com/cosmos/cosmos-sdk/x/crisis/types"
+	distribution "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	"github.com/cosmos/cosmos-sdk/x/feegrant"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	slashingtypes "github.com/cosmos/cosmos-sdk/x/slashing/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
```
Comment (rootulp) [inline app/consistent_apphash_test.go:32]:
```suggestion
```

File: app/consistent_apphash_test.go [hunk 4]
```diff
@@ -3,134 +3,303 @@ package app_test
...
 type BlobTx struct {
 	author    string
 	blobs     []*tmproto.Blob
 	txOptions []user.TxOption
 }
 
-// TestConsistentAppHash executes a set of txs, generates an app hash,
-// and compares it against a previously generated hash from the same set of transactions.
+// TestConsistentAppHash executes all state machine messages, generates an app hash and a data root,
+// and compares it against a previously generated hashes from the same set of transactions.
```
Comment (rootulp) [inline app/consistent_apphash_test.go:48]:
```suggestion
// and compares it against previously generated hashes from the same set of transactions.
```

**Review by evan-forbes — APPROVED** [review]
doesn't have to be changed here, but I still think that pursuing a table driven approach is best here, even if that means we have to add some functionality in testapp

the main reasons were similar to what we've discussed before, just that adding things to table driven tests is intuitive and we can isolate msgs. atm this test can break and we don't immediately know which module changed its state machine in a breaking way

we also don't have the ability to change the app version, which presumably is something else that we would need if we want to make it easy on ourselves in the future when we expand these tests

File: app/consistent_apphash_test.go [hunk 5]
```diff
@@ -3,134 +3,303 @@ package app_test
...
 
-	// Create Blob Tx
-	rawBlobTx, _, err := signer.CreatePayForBlobs(blobTx.author, blobTx.blobs, blobTx.txOptions...)
+	// NewMsgWithdrawDelegatorReward - withdraws delegator rewards
+	msgWithdrawDelegatorReward := distribution.NewMsgWithdrawDelegatorReward(accountAddresses[0], genValidators[0].GetOperator())
+	secondBlockSdkMsgs = append(secondBlockSdkMsgs, msgWithdrawDelegatorReward)
+
+	// ------------ Third Block ------------
+
+	// Txs within the third block are signed by the validator's signer
```
Comment (rootulp) [inline app/consistent_apphash_test.go:250]: > are signed by the validator's signer

is this unique to the third block? If not, why describe it here and not for blocks one and two?

File: app/consistent_apphash_test.go [hunk 6]
```diff
@@ -172,16 +341,119 @@ func deterministicKeyRing(cdc codec.Codec) (keyring.Keyring, []types.PubKey) {
...
+
+	// Deliver SDK Txs
+	for i, tx := range encodedSdkTxs {
+		resp := testApp.DeliverTx(abci.RequestDeliverTx{Tx: tx})
+		if resp.Code != abci.CodeTypeOK {
+			return nil, nil, fmt.Errorf("DeliverTx failed for the message at index %d: %s", i, resp.Log)
+		}
+	}
+
+	// Deliver Blob Txs
```
Comment (rootulp) [inline app/consistent_apphash_test.go:433]:
```suggestion
	// Deliver Blob Tx
```

**Review by rootulp — APPROVED** [review]

File: app/consistent_apphash_test.go (hunk 5 above)
Comment (ninabarbakadze) [reply app/consistent_apphash_test.go:250]: it is unique to the third block

**Comment by ninabarbakadze** [comment]
> doesn't have to be changed here, but I still think that pursuing a table driven approach is best here, even if that means we have to add some functionality in testapp
> 
> the main reasons were similar to what we've discussed before, just that adding things to table driven tests is intuitive and we can isolate msgs. atm this test can break and we don't immediately know which module changed its state machine in a breaking way
> 
> we also don't have the ability to change the app version, which presumably is something else that we would need if we want to make it easy on ourselves in the future when we expand these tests

I'd say that I can revisit this when i have to write consistent apphash test for v2. We can then figure out how we want different versions to be tested + the structure. Then I'd also refactor this to the table driven approach. Updated the AC on the v2 testing issue: https://github.com/celestiaorg/celestia-app/issues/3624

Although the way i see this test breaking mostly is by application producing a different hash to what's expected because we accidentally made an apphash breaking change to the state machine.

File: app/consistent_apphash_test.go [hunk 7]
```diff
@@ -197,6 +466,44 @@ func queryAccountInfo(capp *app.App, accs []string, kr keyring.Keyring) []blobfa
...
+	rec, err := kr.Key(account)
+	if err != nil {
+		panic(err)
+	}
+	addr, err := rec.GetAddress()
+	if err != nil {
+		panic(err)
+	}
+	return addr
+}
```
Comment (cmwaters) [inline app/consistent_apphash_test.go:479]: I thought we already had a function like this in test util package

**Review by cmwaters — APPROVED** [review]

**Review by evan-forbes — APPROVED** [review]

File: app/consistent_apphash_test.go (hunk 7 above)
Comment (ninabarbakadze) [reply app/consistent_apphash_test.go:479]: on main we do

**Pull request merged**